x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/Watcher.java

```diff
@@ -282,6 +282,16 @@ public class Watcher extends Plugin implements SystemIndexPlugin, ScriptPlugin,
      * to these old nodes that the mappings are newer than they are.
      */
     private static final String LEGACY_VERSION_FIELD_VALUE = "8.12.0";
+
+    private static final String REINDEX_SCRIPT_FROM_V7 = """
```

Comment:
This change here is only to illustrate approach. I'll revert it later and create a separate PR for 8.x branch after this PR is backported there. The script will only be in 8.18, not in 9.0